src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -77,6 +79,16 @@ public class ForwardGenerator extends AbstractGenerator {
   /** How to select the method to use for creating a new sequence. */
   private final TypedOperationSelector operationSelector;
 
+  // At most one of the CMSelectors is non-null.
+  /** How to select a constant value that is extracted by Constant Mining in the CLASS level. */
+  private ConstantMiningSelector<ClassOrInterfaceType> classCMSelector;
+
+  /** How to select a constant value that is extracted by Constant Mining in the PACKAGE level. */
+  private ConstantMiningSelector<Package> packageCMSelector;
+
+  /** How to select a constant value that is extracted by Constant Mining in the ALL level. */
+  private TfIdfSelector generalCMSelector;
```

Comment:
I think you can use a single variable, not three, similarly to how there is now just one ConstantMiningStatistics value.